### PR TITLE
Style fixes on the previous merge request

### PR DESCRIPTION
**Pull request by jopedroliveira**

### CODE REVIEW

**Comment by chenguoguo** [comment]
Thanks. I added some comments.  A high level suggestion: could we make this demo reading from raw audio file only, instead of making it for reading from both file and sound card?

The idea is that we already have demo.cc that reads from sound card, so we probably don't want to include it in demo2.cc

**Comment by jopedroliveira** [comment]
@chenguoguo Yes, we can do that. 
I'll review the code under that perspective then I post a new merge.

**Comment by jopedroliveira** [comment]
Ok, @chenguoguo 
Just reviewed the code and removed everything with the microphone audio acquisition and added a new feature. Now we can also read audio from a wav file.

**Comment by jopedroliveira** [comment]
@chenguoguo meanwhile I noted the wav header of snowboy file had an error and fixed it. Also added a .RAW file. You can test the program with both snowboy.wav or snowboy.raw files

File: examples/C++/demo2.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  if (tag[0] != 'R' || tag[1] != 'I' || tag[2] != 'F' || tag[3] != 'F'){
+    fclose(fi);
+    perror("NO 'RIFF'.");
+  }
+  if (wavhdr->fmtTag != 1){
+    fclose(fi);
+    perror("WAV file has encoded data or it is WAVEFORMATEXTENSIBLE.");
+  }
+  if (wavhdr->fmtSize == 14){
+      wavhdr->bps = 16;
```
Comment (chenguoguo) [inline examples/C++/demo2.cpp]: Please fix the indentation spaces here, should be 2.

File: examples/C++/demo2.cpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  }
+  if (wavhdr->fmtSize >= 16){
+      fread(&wavhdr->bps, 2, 1, fi);
```
Comment (chenguoguo) [inline examples/C++/demo2.cpp]: Same as the above, fix the indentation.

File: examples/C++/demo2.cpp [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  }
+  fread(&wavhdr->datasize, 4, 1, fi); //data size
+  // Assuming that header ends here. 
+  // From here until the end it is audio data
+}
+
+
+
+int main(int argc, char * argv[]) {
+ char * usage = (char *) "C++ demo that shows how to use snowboy. \
```
Comment (chenguoguo) [inline examples/C++/demo2.cpp]: Two spaces for indentation. Please fix the code below as well.

File: examples/C++/demo2.cpp [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+ FILE * f;
+ 
+ if(argc > 2 or argc < 2){
+  printf("%s\n",usage);
+  exit(1);
+ }
+
+ if(argc > 1){
+  filename = argv[1]; //input: file. else, input: mic
+ }
```
Comment (chenguoguo) [inline examples/C++/demo2.cpp]: You already made sure argc == 2 above. So you don't need this if.

File: examples/C++/demo2.cpp [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+ if (str.find(type) != std::string::npos) {
+  isRaw = false;
+ }
+
+
+ if(filename != NULL)
+  f = fopen(filename,"rb");
+  
+  if( f == NULL ){
```
Comment (chenguoguo) [inline examples/C++/demo2.cpp]: Don't use space around `f == NULL`

**Comment by jopedroliveira** [comment]
@chenguoguo how's the merge going?

**Comment by jopedroliveira** [comment]
@chenguoguo I already addresed those issues 😄 
check [the last commit](https://github.com/Kitt-AI/snowboy/pull/291/commits/9bcda9dedf1140a5b4b684d07bef91726d6f9f21)

**Comment by chenguoguo** [comment]
OK I'm going to merge it and fix those issues by myself. Some of the comments have not been addressed.

**Pull request merged**